librz/arch/isa/msp430/msp430_il_flags.inc

```diff
@@ -0,0 +1,247 @@
+}
+
+RZ_OWN RzILOpEffect *set_sxt_flags(const char *result_name) {
+	// n z as usual
+	RzILOpBitVector *nz = update_sr_nz_flags(VARL(result_name), update_sr_clear_vcnz(MSP430_GETR(MSP430_SR)));
+	// c as not if result is zero
+	RzILOpBitVector *cnz = update_sr_c_flag(INV(IS_ZERO(VARL(result_name))), nz);
+
+	// v as zero
+	// no need to actually do this, v is already cleared from update_sr_clear_vcnz
+
+	return MSP430_SETR(MSP430_SR, cnz);
+}
+
+RZ_OWN RzILOpEffect *set_and_flags(const char *result_name) {
+	return set_sxt_flags(result_name);
+}
+
+RZ_OWN RzILOpEffect *set_xor_flags(const char *source_name, const char *destination_name, const char *result_name) {
+	// n z as usual
+	RzILOpBitVector *nz = update_sr_nz_flags(VARL(result_name), update_sr_clear_vcnz(MSP430_GETR(MSP430_SR)));
+	// c as not if result is zero
+	RzILOpBitVector *cnz = update_sr_c_flag(INV(IS_ZERO(VARL(result_name))), nz);
+	// v as if both operands are negative
+	RzILOpBitVector *vcnz = update_sr_v_flag(AND(MSB(VARL(source_name)), MSB(VARL(destination_name))), cnz);
+
+	return MSP430_SETR(MSP430_SR, vcnz);
+}
+
+RZ_OWN RzILOpEffect *set_add_flags(const char *source_name, const char *destination_name, const char *result_name) {
+	// n z as usual
+	RzILOpBitVector *nz = update_sr_nz_flags(VARL(result_name), update_sr_clear_vcnz(MSP430_GETR(MSP430_SR)));
+	// c especially for the add
+	RzILOpBitVector *cnz = update_sr_c_flag_add(VARL(source_name), VARL(destination_name), VARL(result_name), nz);
+	// v especially for the add
+	RzILOpBitVector *vcnz = update_sr_v_flag_add(VARL(source_name), VARL(destination_name), VARL(result_name), cnz);
+
+	return MSP430_SETR(MSP430_SR, vcnz);
+}
+
+RZ_OWN RzILOpEffect *do_set_inc_flags(const char *result_name, const char *old_destination_name, RzILOpBitVector *zero_carry_const, RzILOpBitVector *overflow_const) {
+	// n as usual
+	RzILOpBool *is_negative = MSB(VARL(result_name));
+	RzILOpBitVector *n = update_sr_n_flag(is_negative, update_sr_clear_vcnz(MSP430_GETR(MSP430_SR)));
+
+	// z c v by comparison to known constants
+	RzILOpBool *is_zero_and_carry = EQ(VARL(old_destination_name), zero_carry_const);
+	RzILOpBool *is_overflow = EQ(VARL(old_destination_name), overflow_const);
+
+	RzILOpBitVector *nzc = LOGOR(ITE(is_zero_and_carry, U16(0x0003), U16(0)), n);
+	RzILOpBitVector *vnzc = update_sr_v_flag(is_overflow, nzc);
+
+	return MSP430_SETR(MSP430_SR, vnzc);
+}
+
+RZ_OWN RzILOpEffect *set_incb_flags(const char *result_name, const char *old_destination_name) {
+	return do_set_inc_flags(result_name, old_destination_name, U8(0xFF), U8(0x7F));
+}
+
+RZ_OWN RzILOpEffect *set_inc_flags(const char *result_name, const char *old_destination_name) {
+	return do_set_inc_flags(result_name, old_destination_name, U16(0xFFFF), U16(0x7FFF));
+}
+
+RZ_OWN RzILOpEffect *do_set_dec_flags(const char *result_name, const char *old_destination_name, RzILOpBitVector *zero_const, RzILOpBitVector *carry_const, RzILOpBitVector *overflow_const) {
```

Comment:
Please mark all `RzILOpBitVector` parameters of those helper functions as `RZ_OWN` (if they are not duplicated before use, then `RZ_BORROW` of course).